src/lib/__tests__/insights-client-test.ts

```diff
@@ -0,0 +1,261 @@
+      } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toEqual(
+        expect.objectContaining({
+          instantSearchInstance,
+          results,
+          widgetParams,
+        })
+      );
+    });
+
+    it('should expose the insights client wrapper to renderOptions if passed to instantSearchInstance', () => {
+      const { renderFn } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+
+    it('should not expose the insights client wrapper to renderOptions if not passed to instantSearchInstance', () => {
+      const { renderFn } = setup({ insightsClient: undefined });
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+  });
+
+  describe('exposed insights client wrapper', () => {
+    beforeEach(() => {});
+
+    it('should call the insights client under the hood', () => {
+      const { renderFn, instantSearchInstance } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      renderProps.insights('clickedObjectIDsAfterSearch', {
+        objectIDs: ['3'],
+        eventName: 'Add to basket',
+      });
+      expect(instantSearchInstance.insightsClient).toHaveBeenCalledTimes(1);
+    });
+
+    it('should pass it the correct parameters', () => {
+      const { renderFn, instantSearchInstance } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      renderProps.insights('clickedObjectIDsAfterSearch', {
+        objectIDs: ['3'],
+        eventName: 'Add to basket',
+      });
+      const [
+        method,
+        payload,
+      ] = instantSearchInstance.insightsClient.mock.calls[0];
+      expect(method).toEqual('clickedObjectIDsAfterSearch');
+      expect(payload.eventName).toEqual('Add to basket');
+      expect(payload.index).toEqual('theIndex');
+      expect(payload.queryID).toEqual('theQueryID');
+      expect(payload.objectIDs).toEqual(['3']);
+      expect(payload.positions).toEqual([11]);
+    });
+
+    it('should not infer or pass the positions if method is `convertedObjectIDsAfterSearch`', () => {
+      const { renderFn, instantSearchInstance } = setup();
+      const [renderProps] = renderFn.mock.calls[0];
+      renderProps.insights('convertedObjectIDsAfterSearch', {
+        objectIDs: ['1'],
+        eventName: 'Add to basket',
+      });
+      const [
+        method,
+        payload,
+      ] = instantSearchInstance.insightsClient.mock.calls[0];
+      expect(method).toEqual('convertedObjectIDsAfterSearch');
+      expect(payload.eventName).toEqual('Add to basket');
+      expect(payload.index).toEqual('theIndex');
+      expect(payload.queryID).toEqual('theQueryID');
+      expect(payload.objectIDs).toEqual(['1']);
+      expect(payload.positions).toBeUndefined();
```

Comment:
We could write it with one `toEqual` assertions, it avoid to create assertions for the negation.

```
expect(payload).toEqual({
  eventName: "Add to basket",
  index: "theIndex",
  queryID: "theQueryID",
  objecIDs: ["1"]
});
```

Reply:
fixed in 1de2de18f24bee9945916d3dc1ea9652eeb17773